src/player.cpp

```diff
@@ -3232,19 +3232,21 @@ void player::update_stomach( const time_point &from, const time_point &to )
     const bool mycus = has_trait( trait_M_DEPENDENT );
     const float kcal_per_time = get_bmr() / ( 12.0f * 24.0f );
     const int five_mins = ticks_between( from, to, 5_minutes );
+    const int half_hours = ticks_between( from, to, 30_minutes );
 
     if( five_mins > 0 ) {
-        stomach.absorb_water( *this, 250_ml * five_mins );
-        guts.absorb_water( *this, 250_ml * five_mins );
-    }
-    if( ticks_between( from, to, 30_minutes ) > 0 ) {
-        // the stomach does not currently have rates of absorption, but this is where it goes
-        stomach.calculate_absorbed( stomach.get_absorb_rates( true, rates ) );
-        guts.calculate_absorbed( guts.get_absorb_rates( false, rates ) );
-        stomach.store_absorbed( *this );
-        guts.store_absorbed( *this );
-        guts.bowel_movement( guts.get_pass_rates( false ) );
-        stomach.bowel_movement( stomach.get_pass_rates( true ), guts );
+        // Digest nutrients in stomach, they are destined for the guts (except water)
+        nutrients digested_to_guts = stomach.digest( *this, five_mins, half_hours );
+        // Digest nutrients in guts, they will be distributed to needs levels
+        nutrients digested_to_body = guts.digest( *this, five_mins, half_hours );
+        // Water from stomach skips guts and gets absorbed by body
+        set_thirst( std::max(
+                        -100, get_thirst() - units::to_milliliter<int>( digested_to_guts.water ) / 5 ) );
+        guts.ingest( digested_to_guts );
+        if( !is_npc() || !get_option<bool>( "NO_NPC_FOOD" ) ) {
```

Comment:
```suggestion
        if( !( is_npc() && get_option<bool>( "NO_NPC_FOOD" ) ) ) {
```

IMHO this is more readable.  "If it's not an NPC and we're using no NPC food" as opposed to "if it's not an NPC or we're _not_ using _no_ NPC food".

Reply:
> "If it's not an NPC and we're using no NPC food"

No, this isn't a correct articulation of your version. That wording would represent
```cpp
if( !is_npc() && get_option<bool>( "NO_NPC_FOOD" ) ) {
```
which isn't what we want.

Reply:
Yeah the parenthesis don't come across in the word version I wrote earlier.  My code is more like "[Unless](https://ruby-doc.org/core-2.6.5/doc/syntax/control_expressions_rdoc.html#label-unless+Expression) it's an NPC and we're using No NPC Food".  Instead of a comment it could move to a `should_process_food` method and then `if( should_process_food() ) {`.

Reply:
I still have to disagree, I think my version is more readable, and I don't want to add a function to a class we're trying to deprecate. I did add a comment though. I'll leave this conversation unresolved though so whoever goes to merge this can let me know if they think a change needs to be made.